cg/services/orders/store_order_services/store_generic_order.py

```diff
@@ -1,99 +1,40 @@
-        for sample in samples:
-            subject_id: str = sample.subject_id
-            if not subject_id:
-                continue
-            new_gender: str = sample.sex
-            if new_gender == "unknown":
-                continue
-
-            existing_samples: list[Sample] = self.status.get_samples_by_customer_and_subject_id(
-                customer_internal_id=customer_id, subject_id=subject_id
-            )
-            existing_sample: Sample
-            for existing_sample in existing_samples:
-                previous_gender = existing_sample.sex
-                if previous_gender == "unknown":
-                    continue
-
-                if previous_gender != new_gender:
-                    raise OrderError(
-                        f"Sample gender inconsistency for subject_id: {subject_id}: previous gender {previous_gender}, new gender {new_gender}"
-                    )
-
-    def _validate_samples_available_to_customer(
-        self, samples: list[OrderInSample], customer_id: str
-    ) -> None:
-        """Validate that the customer have access to all samples"""
-        sample: Of1508Sample
-        for sample in samples:
-            if not sample.internal_id:
-                continue
-
-            existing_sample: Sample = self.status.get_sample_by_internal_id(
-                internal_id=sample.internal_id
-            )
-
-            data_customer: Customer = self.status.get_customer_by_internal_id(
-                customer_internal_id=customer_id
-            )
-
-            if existing_sample.customer not in data_customer.collaborators:
-                raise OrderError(f"Sample not available: {sample.name}")
-
-    def _validate_case_names_are_unique(
-        self, samples: list[OrderInSample], customer_id: str
-    ) -> None:
-        """Validate that the names of all cases are unused for all samples"""
-
-        customer: Customer = self.status.get_customer_by_internal_id(
-            customer_internal_id=customer_id
-        )
-
-        sample: Of1508Sample
-        for sample in samples:
-            if self._is_rerun_of_existing_case(sample=sample):
-                continue
-            if self.status.get_case_by_name_and_customer(
-                customer=customer, case_name=sample.family_name
-            ):
-                raise OrderError(f"Case name {sample.family_name} already in use")
-
-    def submit_order(self, order: OrderIn) -> dict:
+class StoreGenericOrderService(StoreOrderService):
+    """
+    Service for storing generic orders in StatusDB and Lims.
+    This class is used to store orders for the following workflows:
+    - Balsamic
+    - Balsamic QC
+    - Balsamic UMI
+    - MIP DNA
+    - MIP RNA
+    - Tomte
+    """
+
+    def __init__(
+        self,
+        status_db: Store,
+        lims_service: OrderLimsService,
+    ):
+        self.status_db = status_db
+        self.lims = lims_service
+
+    def store_order(self, order: OrderIn) -> dict:
```

Comment:
Why does a `store` function return a dict? This is a bit confusing, maybe the storing should be one function and another "getter" function should be implemented if an order in a dict form is required

Reply:
The dict return object is one of the most confusing objects that I have seen and should definitely be rewritten at some point. Especially because the structure of the dict depends on the workflow. Probably not in scope though?

Reply:
Yes I agree, right now it returns something which should not be in the future. It should just take a model, translate it into database models and store the stuff. Yet, currently the submitters all return a dict. To not break functionality for now I wanted to extract the existing logic as is and improve upon it later on.